iceoryx_posh/test/mocks/chunk_mock.hpp

```diff
@@ -50,9 +58,10 @@ class ChunkMock
         {
             m_chunkHeader->~ChunkHeader();
         }
-        if (m_rawMemory != nullptr)
+        if (m_unalignedMemory != nullptr)
```

Comment:
this is weird. You allocate to m_rawMemory and free m_unalignedMemory. How about reverting this and introducing an m_alignedMemory
```
    m_rawMemory = static_cast<uint8_t*>(malloc(Size + Alignment));
    auto memAddress = reinterpret_cast<uint64_t>(m_rawMemory);
    auto offset = memAddress % Alignment;
    auto alignedAddress =  offset = 0 ? memAddress : memAddress + Alignment - offset;
    m_alignedMemory = reinterpret_cast<uint8_t*>(alignedAddress );
```

This has also the benefit to fix the bugs in the implementation ;)

Reply:
btw, you could also remove the platform defines and use the mac solution for all platforms. This reduces the complexity of the mock

Reply:
done